src/app/rpmostree-builtin-status.cxx

```diff
@@ -480,9 +480,11 @@ print_daemon_state (RPMOSTreeSysroot *sysroot_proxy, GCancellable *cancellable,
   return TRUE;
 }
 
-/* Print the result of rpmostree_context_get_rpmmd_repo_commit_metadata() */
+/* Print the commit checksum, and optionally the rpm repodata injected from
+ * rpmostree_context_get_rpmmd_repo_commit_metadata() */
 static void
-print_origin_repos (gboolean host_endian, guint maxkeylen, GVariantDict *commit_meta)
+print_commit (const char *key, const char *checksum, gboolean host_endian, guint maxkeylen,
```

Comment:
I'm finding this commit hard to review/think about in detail.

How much have you played with this and looked at the output?  Hmm...I wonder how hard it'd be to wire up unit tests (or just a manual script) that take a set of origin files  and outputs `rpm-ostree status`.